File: backend/models/Book.js
// backend/models/Book.js
const mongoose = require('mongoose');

const BookSchema = new mongoose.Schema({
  title: { type: String, required: true },
  author: { type: String, required: true },
  description: { type: String },
  cover_image: { type: String }, // Link ảnh bìa
  price: { type: Number, required: true }, // Giá tiền
  type: { 
    type: String, 
    enum: ['ebook', 'physical', 'comic'], 
    required: true 
  },
  // Nếu là ebook thì cần link tải
  file_url: { type: String, default: '' },
  allow_download: { type: Boolean, default: false },
  
  // Thống kê
  total_reads: { type: Number, default: 0 },
}, { timestamps: true });

module.exports = mongoose.model('Book', BookSchema);

File: book_app_mobile/lib/models/book.dart
class Book {
  final String id;
  final String title;
  final String author;
  final int price;
  final String coverImage;

  Book({
    required this.id,
    required this.title,
    required this.author,
    required this.price,
    required this.coverImage,
  });

  // Hàm này giúp chuyển đổi cục JSON loằng ngoằng thành Object gọn gàng
  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['_id'], // MongoDB dùng _id
      title: json['title'] ?? 'Không tên',
      author: json['author'] ?? 'Ẩn danh',
      price: json['price'] ?? 0,
      coverImage: json['cover_image'] ?? '',
    );
  }
}